estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,56 @@
+from datetime import timedelta
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+
+class EstatePropertyType(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Offers for real state properties'
+
+    validity = fields.Integer(string="Validity (Days)",default=7)
+
+
+    price = fields.Float()
+    status = fields.Selection(
+        selection=[("accepted", "Accepted"), ("refused", "Refused")], copy=False
+    )
+    property_id = fields.Many2one(comodel_name='estate.property')
+    partner_id = fields.Many2one(comodel_name='res.partner', string='Buyer')
+
+
+    date_deadline = fields.Date(compute='_compute_date_deadline', inverse='_inverse_date_deadline')
+    
+
+
+    _sql_constraints = [
+        ('check_price', 'CHECK(price >= 0)',
+         'Offer prices MUST be postive.'),
+
+    ]
+
+    def action_accept(self):
+        for record in self:
+            try:
+                record.property_id.action_process_accept(self)
+                record.status = 'accepted'
+            except UserError as e:
+                raise e
```

Comment:
Only raise builtin odoo errors for optimal UX
```suggestion
                raise UserError("error message")
```